src-exe/cabal-plan.hs

```diff
@@ -106,18 +134,97 @@ doInfo (plan,projbase) = do
     let locals = [ Unit{..} | Unit{..} <- M.elems pm, uType == UnitTypeLocal ]
         pm = pjUnits plan
 
-    forM_ locals $ \pitem -> do
+    for_ locals $ \pitem -> do
         print (uPId pitem)
-        forM_ (M.toList $ uComps pitem) $ \(ct,ci) -> do
+        for_ (M.toList $ uComps pitem) $ \(ct,ci) -> do
             print ct
-            forM_ (S.toList $ ciLibDeps ci) $ \dep -> do
+            for_ (S.toList $ ciLibDeps ci) $ \dep -> do
                 let Just dep' = M.lookup dep pm
                     pid = uPId dep'
                 putStrLn ("  " ++ T.unpack (dispPkgId pid))
             putStrLn ""
 
     return ()
 
+doDot :: Bool -> Bool -> PlanJson -> IO ()
+doDot showBuiltin showGlobal plan = do
+    putStrLn "digraph plan {"
```

Comment:
Alga has built-in `Algebra.Graph.Export.Dot` which you might find useful:

http://hackage.haskell.org/package/algebraic-graphs-0.0.5/docs/Algebra-Graph-Export-Dot.html#v:export

Reply:
Yes, I'd used that, if not the version restrictions on *alga* atm. (That code is part of already merged PR).